spec/02-integration/05-proxy/14-upstream-status-header_spec.lua

```diff
@@ -0,0 +1,142 @@
+local helpers = require "spec.helpers"
+local constants = require "kong.constants"
+
+describe(constants.HEADERS.UPSTREAM_STATUS .. " header", function()
+  local client
+
+  setup(function()
+    local bp = helpers.get_db_utils()
+    assert(helpers.dao:run_migrations())
+
+    local service = bp.services:insert {
+      host = helpers.mock_upstream_host,
+      port = helpers.mock_upstream_port,
+      protocol = helpers.mock_upstream_protocol,
+    }
+    assert(service)
+
+    local route1 = bp.routes:insert {
+      protocols = { "http" },
+      service = service,
+      paths = { "/foo" },
+    }
+    assert(route1)
+
+    local route2 = bp.routes:insert {
+      protocols = { "http" },
+      service = service,
+      paths = { "/bar" },
+    }
+    assert(route2)
```

Comment:
The Lua idiom we follow is to wrap the `:insert()` call within an `assert()`. The new DAO is built for this purpose in mind, since the second return value will be a string, thus producing the equivalent of `assert(nil, err_from_insert)`. This is a typical use of assert and errors with Lua. In the current case, the assert would fail but no errors would be printed explaining _why_. It's also shorter and more eleguant.

Reply:
Thanks for explaining this, I'll keep in mind next time. Updated this in the most recent push.